mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -62,20 +60,25 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter, IReplaceableNode {
         }
 
     override fun replaceNode(concept: ConceptReference?): INode {
-        requireNotNull(concept) { "Can't replace $node with null concept. Use BaseConcept explicitly." }
+        requireNotNull(concept) { "Cannot replace node `$node` with a null concept. Explicitly specify a concept (e.g., `BaseConcept`)." }
+        val parent = requireNotNull(node.parent) { "Cannot replace node `$node` because it has no parent." }
```

Comment:
This should be allowed. A node can be a root node of a model, in which case we can use `model.addRootNode` instead of `parent.insertChildBefore`, or it can be a "free floating" node and then we just don't have to update the containment relation.